operators/to_mapbox/__init__.py

```diff
@@ -0,0 +1,152 @@
+
+        # LOCATION PRE-PROCESSING
+        DF.update_resource(['Locations'], **{'name': 'locations', 'path': 'locations.csv'}),
+        DF.filter_rows(lambda r: any(
+            all(r.get(f) for f in fields)
+            for fields in [('resolved_lat', 'resolved_lon'), ('fixed_lat', 'fixed_lon')]
+        ), resources=['locations']),
+        DF.add_field('lat', 'number', lambda r: r.get('fixed_lat') or r['resolved_lat'], resources=['locations']),
+        DF.add_field('lon', 'number', lambda r: r.get('fixed_lon') or r['resolved_lon'], resources=['locations']),
+        DF.add_field('geometry', 'geopoint', lambda r: [r['lon'], r['lat']], resources=['locations']),
+        DF.add_field('address', 'string', lambda r: r.get('resolved_address') or r['id'], resources=['locations']),
+        DF.select_fields(['pk', 'geometry', 'address'], resources=['locations']),
+
+        # JOIN TO PRODUCE MAPBOX DATA
+        DF.duplicate('services', 'to_mapbox', 'to_mapbox.csv'),
+        DF.rename_fields({'name': 'service_name'}, resources=['to_mapbox']),
+        DF.join(
+            'locations', ['pk'], 'branches', ['location_pk'],
+            fields={'geometry': {'name': 'geometry', 'aggregate': 'last'}, 'address': {'name': 'address', 'aggregate': 'last'}},
+            source_delete=False,
+        ),
+        DF.join(
+            'organizations', ['pk'], 'branches', ['organization_pk'],
+            fields={'organization_name': {'name': 'name', 'aggregate': 'last'}},
+            source_delete=False,
+        ),
+        DF.join(
+            # was assuming branch names could differ from org names,
+            # but from reviewing the data looks like not.
+            # bringing both over anyway because branch_name None means org offers service with no branch.
+            'branches', ['organization_pk'], 'to_mapbox', ['organization_pk'],
+            fields={
+                'branches': {'name': 'name', 'aggregate': 'set'},
+                'organizations': {'name': 'organization_name', 'aggregate': 'set'},
+                'geometry': {'name': 'geometry', 'aggregate': 'last'},
+                'address': {'name': 'address', 'aggregate': 'last'},
+            },
+            source_delete=False,
+        ),
+        DF.join(
+            'responses', ['pk'], 'to_mapbox', ['response_pk'],
+            fields={'response_id': {'name': 'id', 'aggregate': 'last'}, 'response_name': {'name': 'name', 'aggregate': 'last'}},
+            source_delete=False
+        ),
+
+        # TODO - geometry can still be null. I guess it needs to be filtered out as this data is *only* for the map.
+
+        DF.add_field('branch_name', 'string', resources=['to_mapbox']),
+        unwind('branches', 'branch_name', resources=['to_mapbox']),
+        DF.add_field('organization_name', 'string', resources=['to_mapbox']),
+        unwind('organizations', 'organization_name', resources=['to_mapbox']),
```

Comment:
wouldn't this create a copy of each branch name for each organization? 
e.g. the service was tied to (Branch_A/Org_A) & (Branch_B/Org_B), it will first unwind to
Branch_A, (Org_A, Org_B)
Branch_B, (Org_A, Org_B)

and then unwind the organizations, creating a record for (Branch_A, Org_B) etc.

Reply:
Yes, the logic is incorrect.